### ISSUE TITLE
Kundera Cassandra - Inheritance not working when using collection in Parent class

### ISSUE THREAD

**Issue by pmanicka** [Impetus/kundera#866]
Hi,

We are trying to use Kundera for our Cassandra NoSql and we are using Inheritence with Discriminator Column. In the parent class, there is a column which of type Map. When trying to persist in cassandra using the child objects, its throwing error like "why:Multiple definition of identifier " 

21:03:00.038 [main] INFO  c.i.k.configure.SchemaConfiguration - Configuring schema export for : cassandra_putest
21:03:00.233 [main] INFO  c.i.c.c.c.CassandraPropertyReader - Returning placement strategy value org.apache.cassandra.locator.SimpleStrategy
21:03:00.746 [main] ERROR c.i.c.c.s.CassandraSchemaManager - Error occurred while creating table TABLEA, Caused by: InvalidRequestException(why:Multiple definition of identifier PROPERTIES).

Sample code below.

@Entity
@Table(name="TABLEA")
@Inheritance(strategy=InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="TYPE", discriminatorType=DiscriminatorType.STRING)
@IndexCollection(columns = { @Index(name = "ID"), @Index(name = "PARENT_REFERENCE_ID")})
public abstract class A{

@Column(name = "ID", nullable = false)
    protected String id;

```
@ElementCollection
@Column(name = "PROPERTIES")
private Map<String, String> Properties;
```

}

@Entity
@Table(name="B")
@DiscriminatorValue("B")
public class Feeder extends A{

}

@Entity
@Table(name="C")
@DiscriminatorValue("C")
public class Feeder extends A{

}

**Comment by devender-yadav** [comment]
Hi @pmanicka,

I am able to replicate your issue at my end. We will fix this issue in the next release (_expected to be released in mid June_).

-Dev

**Comment by pmanicka** [comment]
Hi Devender,

Thanks. One more question is, Is Kundera jars are backward compatible. We are using JDK 1.5 and its failing with below error during maven build.
com\impetus\kundera\core\kundera-core\3.3\kundera-core-3.3.jar(com/impetus/kundera/index/Index.class)
class file has wrong version 51.0, should be 49.0

**Comment by devender-yadav** [comment]
@pmanicka 

Kundera from 2.12 release onwards has become JPA2.1 compliant whose one of the per-requisite is JDK1.7. So, newer versions can not be backward compatible.

-Dev

**Comment by karthikprasad13** [comment]
@pmanicka 

Added fix for this issue, you can build the code from trunk or wait till next release. Also, Do not use `@ElementCollection` annotation on Collection of simple datatypes `(Map<String, Sring>)`. It is used on Collections of Embeddable type `(Map<String, Address>)`.

-Karthik

**Comment by devender-yadav** [comment]
@pmanicka 

Fixed in Kundera 3.5

-Dev
